lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -424,7 +424,7 @@ def cleanup():
         ignore_lock = (
             "data" in deployment_desc.config.roles
```

Comment:
1. should this not also check if the roles are autogenerated? given that has an implicit `data` role.
2. How do we guarantee that this only runs for the `first data node joining`?